providers/rbs/api/client.go

```diff
@@ -0,0 +1,159 @@
+// Copyright (c) Facebook, Inc. and its affiliates.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package api
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"log"
+	"net/http"
+	"net/url"
+	"sync"
+	"time"
+
+	"golang.org/x/oauth2"
+	"golang.org/x/oauth2/clientcredentials"
+
+	"github.com/facebookincubator/nvdtools/providers/lib/download"
+	"github.com/facebookincubator/nvdtools/providers/rbs/schema"
+)
+
+const (
+	pageSize = 100
+)
+
+type Client struct {
+	client    *http.Client
+	baseURL   string
+	userAgent string
+}
+
+func NewClient(clientID, clientSecret, tokenURL, baseURL, userAgent string) (*Client, error) {
+	httpClient, err := download.Client()
+	if err != nil {
+		return nil, fmt.Errorf("couldn't obtain http client: %v", err)
+	}
+
+	ctx := context.WithValue(oauth2.NoContext, oauth2.HTTPClient, httpClient)
+
+	conf := &clientcredentials.Config{
+		ClientID:     clientID,
+		ClientSecret: clientSecret,
+		TokenURL:     tokenURL,
+	}
+
+	c := Client{
+		client:    conf.Client(ctx),
+		baseURL:   baseURL,
+		userAgent: userAgent,
+	}
+
+	return &c, nil
+}
+
+func (c *Client) FetchAllVulnerabilitiesAfterVulndbID(vulndbID int) (<-chan *schema.Vulnerability, error) {
```

Comment:
super-nitpick, but i don't like all these [Ff]etchAllVulnerabilities.* names. I'd dropped the 'Vulnerabilities' part.

Reply:
Will make it uniform in my next diffs